### PR TITLE
[GLUTEN-8005][VL] Add MergeTwoPhasesHashBaseAggregate to injectRas list

### PR DESCRIPTION
**Pull request by yikf**
## What changes were proposed in this pull request?
This PR aims to add MergeTwoPhasesHashBaseAggregate to injectRas list to make RAS support merge aggregate, and enable some UTs related to merge aggregate.

## How was this patch tested?
Exist UTs.

This follows https://github.com/apache/incubator-gluten/pull/7752.

### CODE REVIEW

**Comment by yikf** [comment]
@zhztheplayer @jackylee-ch Could you please take a look, thanks

File: gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala [hunk 1]
```diff
@@ -843,7 +843,6 @@ class ClickHouseTestSettings extends BackendTestSettings {
     .exclude("do not replace hash aggregate if child does not have sort order")
     .exclude("do not replace hash aggregate if there is no group-by column")
     .excludeGlutenTest("replace partial hash aggregate with sort aggregate")
-    .excludeGlutenTest("replace partial and final hash aggregate together with sort aggregate")
```
Comment (jackylee-ch) [inline gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala:846]: why remove this? I don't see any change about ch backend.

Comment (yikf) [reply gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala:846]: before, the CH backend exclude this UT, but now it is actually can enable due to https://github.com/apache/incubator-gluten/pull/7752/files#diff-8fd5d5c17d4e0d738c5089d703910bda7d1598728324a2987965d8c60fd5c9bbR89

**Comment by zhztheplayer** [comment]
@yikf should we also remove [this line](https://github.com/apache/incubator-gluten/blame/57085144ee2ca6228dffdf15a1407e02c23ab41a/gluten-ut/test/src/test/scala/org/apache/gluten/execution/MergeTwoPhasesHashBaseAggregateSuite.scala#L56), thanks!

**Comment by yikf** [comment]
> @yikf should we also remove [this line](https://github.com/apache/incubator-gluten/blame/57085144ee2ca6228dffdf15a1407e02c23ab41a/gluten-ut/test/src/test/scala/org/apache/gluten/execution/MergeTwoPhasesHashBaseAggregateSuite.scala#L56), thanks!

yes, thanks

**Review by jackylee-ch — APPROVED** [review]

**Review by zhztheplayer — COMMENTED** [review]
Thanks!

**Pull request merged**